pkg/models/commandvalue.go

```diff
@@ -167,14 +181,26 @@ func NewFloat64Value(ro *models.ResourceOperation, origin int64, value float64)
 //NewCommandValue create a CommandValue according to the Type supplied
 func NewCommandValue(ro *models.ResourceOperation, origin int64, value interface{}, t ValueType) (cv *CommandValue, err error) {
 	cv = &CommandValue{RO: ro, Origin: origin, Type: t}
-	if t != String {
+	if t == Binary {
+		// assign cv.binValue
+		err = encodeBinaryValue(cv, value)
+	} else if t != String {
 		err = encodeValue(cv, value)
 	} else {
 		cv.stringValue = value.(string)
 	}
 	return
 }
 
+// NewBinaryValue creates a CommandValue with binary payload and enforces the memory limit for event readings.
+func NewBinaryValue(ro *models.ResourceOperation, origin int64, value []byte) (cv *CommandValue, err error) {
+	if binary.Size(value) > MaxBinaryBytes {
+		return nil, fmt.Errorf("Requested CommandValue payload exceeds limit for binary readings (%v bytes)", MaxBinaryBytes)
+	}
+	cv = &CommandValue{RO: ro, Origin: origin, Type: Binary, BinValue: value}
+	return
```

Comment:
Should we encode the binary value here?
err = encodeBinaryValue(cv, value)

Reply:
This NewBinaryValue helper is responsible just to add the payload to CommandValue, so it matches expectations alongside the other New**Datatype**Value methods.

Similarly, encoding for this data type is handled conditionally in "NewCommandValue" method based on the assigned type.

Reply:
In this case, should we use NewCommandValue for the simple-driver?  Actually, I am not sure how to choose when I should use which helper function to create binary CommandValue